processor/probabilisticsamplerprocessor/murmur3.go

```diff
@@ -0,0 +1,69 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package probabilisticsamplerprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor"
+
+// hash is a murmur3 hash function, see http://en.wikipedia.org/wiki/MurmurHash
+func hash(key []byte, seed uint32) (hash uint32) {
```

Comment:
Would it be worth important a maintained library here so we are not reimplementing known code?

For example: https://github.com/twmb/murmur3

Reply:
Not sure - here I'm just moving code around, and I don't want to break existing functionality for traces. It might make sense to follow up on this separately.

Reply:
I created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16456 to track this.